inc/toolbox.class.php

```diff
@@ -2666,16 +2666,18 @@ static function convertTagToImage($content_text, CommonDBTM $item, $doc_data = [
                if ($document->getFromDB($id)
                    && strpos($document->fields['mime'], 'image/') !== false) {
                   // append ticket reference in image link
-                  $ticket_url_param = "";
-                  if ($item instanceof Ticket) {
-                     $ticket_url_param = "&tickets_id=".$item->fields['id'];
+                  $itil_url_param = "";
+                  if ($item instanceof CommonITILObject) {
+                     $foreign_key = $item->getForeignKeyField();
+                     $itil_url_param = "&{$foreign_key}={$item->fields['id']}";
```

Comment:
Maybe sthing like:
```
$itil_url_param = "&{$foreign_key}=" . $item->input['_job'] instanceof CommonITILObject) ? 
   $item->fields['id'] :
   $item->input['_job']->fields['id'];
```

Reply:
This method may act on an item that is not related to any ITIL item.
I propose https://github.com/glpi-project/glpi/pull/5472/commits/47f326ecf7e98550e417aeafb99beec694a7b04d to not duplicate string concatenation logic.

Reply:
:+1: